src/cli.rs

```diff
@@ -104,36 +106,73 @@ fn is_value_arg(arg: &str, field: &str) -> Option<ArgKind> {
 fn parse_next_arg<T>(
     arg: String,
     out: &mut Vec<String>,
-    parse: impl Fn(&str) -> T,
+    parse: impl Fn(&str) -> Result<T>,
+    store_cb: impl Fn(String) -> Result<String>,
     iter: &mut impl Iterator<Item = String>,
-) -> Option<T> {
+) -> Result<Option<T>> {
     out.push(arg);
     match iter.next() {
         Some(next) => {
-            let result = parse(&next);
-            out.push(next);
-            Some(result)
+            let result = parse(&next)?;
+            out.push(store_cb(next)?);
+            Ok(Some(result))
         }
-        None => None,
+        None => Ok(None),
     }
 }
 
-fn parse_equal_arg<T>(arg: String, out: &mut Vec<String>, parse: impl Fn(&str) -> T) -> T {
-    let result = parse(arg.split_once('=').expect("argument should contain `=`").1);
-    out.push(arg);
+fn parse_equal_arg<T>(
+    arg: String,
+    out: &mut Vec<String>,
+    parse: impl Fn(&str) -> Result<T>,
+    store_cb: impl Fn(String) -> Result<String>,
+) -> Result<T> {
+    let (first, second) = arg.split_once('=').expect("argument should contain `=`");
+    let result = parse(second)?;
+    out.push(format!("{first}={}", store_cb(second.to_owned())?));
 
-    result
+    Ok(result)
 }
 
-fn parse_manifest_path(path: &str) -> Option<PathBuf> {
+fn parse_manifest_path(path: &str) -> Result<Option<PathBuf>> {
     let p = PathBuf::from(path);
-    env::current_dir().ok().map(|cwd| cwd.join(p))
+    Ok(absolute_path(p).ok())
 }
 
 fn parse_target_dir(path: &str) -> Result<PathBuf> {
     absolute_path(PathBuf::from(path))
 }
 
+fn identity(arg: String) -> Result<String> {
+    Ok(arg)
+}
+
+fn str_to_owned(arg: &str) -> Result<String> {
+    Ok(arg.to_owned())
+}
+
+#[allow(clippy::needless_return)]
+fn store_manifest_path(path: String) -> Result<String> {
+    let p = Path::new(&path);
+    if p.is_absolute() {
+        #[cfg(target_os = "windows")]
+        {
+            return p.as_wslpath();
+        }
+        #[cfg(not(target_os = "windows"))]
+        {
+            use crate::file::ToUtf8;
+            return p.to_utf8().map(ToOwned::to_owned);
+        }
+    } else {
+        p.as_posix()
```

Comment:
We don't force absolute paths on the client, just in case we mount at `/project`, which would break our code. There's probably a better fix down the line, but at least this is much better than nothing.